pandas/src/inference.pyx

```diff
@@ -85,6 +85,53 @@ try:
 except AttributeError:
     pass
 
+
+cdef class Seen(object):
+    """
+    Class for keeping track of the types of elements
+    encountered when trying to perform type conversions.
+    """
+
+    cdef public:
+        bint seen_int
```

Comment:
I would also change these variables to something like

```
_bool
_null
_uint
```

makes reading much easier w/o all the extra seen (alternatively ``sbool, snull, suint, sint, sfloat``.....)